geowebcache/s3storage/src/test/java/org/geowebcache/s3/S3BlobStoreConfigProviderTest.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * @author Mauro Bartolomeoli, GeoSolutions Sas, Copyright 2017
+ */
+package org.geowebcache.s3;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.geowebcache.GeoWebCacheExtensions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+import com.thoughtworks.xstream.XStream;
+
+public class S3BlobStoreConfigProviderTest {
+
+    @Before
+    public void setUp() throws Exception {
+        System.setProperty("BUCKET", "MYBUCKET");
+        System.setProperty("CONNECTIONS", "30");
+        System.setProperty("ENABLED", "true");
+        System.setProperty("ALLOW_ENV_PARAMETRIZATION", "true");
+        ClassPathXmlApplicationContext context = new ClassPathXmlApplicationContext(
+                "appContextTestS3.xml");
+
+        GeoWebCacheExtensions gse = new GeoWebCacheExtensions();
+        gse.setApplicationContext(context);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        System.setProperty("BUCKET", "");
```

Comment:
This sets the system property to the empty strings but does not remove it. To make extra sure to leave nothing behind, best to use [clearProperty](https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#clearProperty(java.lang.String)) instead.